apps/teams-test-app/src/components/AppInitialization.tsx

```diff
@@ -45,12 +45,35 @@ const NotifyFailure = (): React.ReactElement =>
     },
   });
 
+const NotifyExpectedFailure = (): React.ReactElement =>
+  ApiWithTextInput<app.ExpectedFailureReason>({
+    name: 'appInitializationExpectedFailure2',
+    title: 'appInitialization.expectedFailure',
+    onClick: {
+      validateInput: input => {
+        if (!input) {
+          // this API actually allow for the input not to be provided
+          return;
+        }
+        const acceptableValues = Object.values(app.ExpectedFailureReason);
+        if (!acceptableValues.includes(input)) {
+          throw new Error(`input must be one of: ${JSON.stringify(acceptableValues)}`);
+        }
```

Comment:
input should be of type below to align with the API "IExpectedFailureRequest " :

Reply:
I apologize, I'm not sure what you mean by "type below", this is probably very obvious.  I am pulling the values from the IExpectedFailure enum

Reply:
My bad for vague comment above  I did not realize type did not turn up there.
The type of notifyExpectedFailure is :   
export interface IExpectedFailureRequest {
    reason: ExpectedFailureReason;
    message?: string;
  }

Reply:
ok this makes sense after talking offline...